composer/datasets/in_context_learning_evaluation.py

```diff
@@ -926,227 +1007,152 @@ class InContextLearningCodeEvalDataset(Dataset):
+        tokenized_example['language'] = example['language']
+        return tokenized_example
 
     def collate_fn(self, data):
-        inputs, prompts, tests, canonical_solutions, entry_points, test_inputs, test_outputs, languages = [], [], [], [], [], [], [], []
-        for sample in data:
-            preamble, prompt, text_prompt, canonical_solution, test, entry_point, test_input, test_output, language = (
-                sample['preamble'],
-                sample['prompt'],
-                sample['prompt_text'],
-                sample['canonical_solution'],
-                sample['test'],
-                sample['entry_point'],
-                sample['test_inputs'],
-                sample['test_outputs'],
-                sample['language'],
-            )
-            context_enc = preamble['input_ids'] + prompt['input_ids']
-            inp, _ = _make_padded_input(context_enc, [],
-                                        self.max_prompt_length,
-                                        self.pad_tok_id,
-                                        padding_side=self.padding_side)
-
-            inputs.append(inp)
-            tests.append(test)
-            prompts.append(text_prompt)
-            canonical_solutions.append(canonical_solution)
-            entry_points.append(entry_point)
-            test_inputs.append(test_input)
-            test_outputs.append(test_output)
-            languages.append(language)
+        """
+        The function that the dataloader uses to accumulate data into batches
+        Args:
+            data (list): list of tokenized datapoints (dicts returned by self._tokenize_example)
 
+        Returns:
+            dict: dictionary for a single batch
+        """
         batch = {
-            'input_ids': torch.stack(inputs),
+            'input_ids': [],
             'mode': 'generate',
-            'labels': canonical_solutions,
-            'prompts': prompts,  # list of prompts
-            'tests': tests,  # list of tests
-            'canonical_solutions': canonical_solutions,  # list of solutions
-            'entry_points': entry_points,  # list of entry points
-            'test_inputs': test_inputs,  # list of test inputs
-            'test_outputs': test_outputs,  # list of test outputs
-            'languages': languages,  # list of languages
+            'labels': [],
+            'prompts': [],  # list of prompts
+            'tests': [],  # list of tests
+            'canonical_solutions': [],  # list of solutions
+            'entry_points': [],  # list of entry points
+            'test_inputs': [],  # list of test inputs
+            'test_outputs': [],  # list of test outputs
+            'languages': [],  # list of languages
             'pass_at_k': self.pass_at_k,
             'generation_length': self.max_seq_len - self.max_prompt_length,
             'generation_kwargs': {
                 'pad_token_id': self.pad_tok_id,
+                # TODO: specify this?
```

Comment:
is the arg just getting ignored in the gauntlet yaml? https://github.com/mosaicml/llm-foundry/blob/e191b0501a7d1d99790a2218025539c4771082a4/scripts/eval/yamls/coding_tasks.yaml#L7

Reply:
ahh i see